### ISSUE TITLE
Low performance of the lib

### ISSUE THREAD

**Issue by Kolyall** [sharish/ShimmerRecyclerView#12]
Main Thread lags, after making ShimmerAdapter show : `mRecyclerView.showShimmerAdapter()`,

**Comment by veghtomi** [comment]
@Kolyall we also had this kind of issue, we changed the shimmer implementation with this shimmer lib: https://github.com/team-supercharge/ShimmerLayout

**Comment by Kolyall** [comment]
@veghtomi Thanks, I changed ShimmerViewHolder:
```
public class ShimmerViewHolder extends RecyclerView.ViewHolder {

    public ShimmerViewHolder(LayoutInflater inflater, ViewGroup parent, int innerViewResId) {
        super(inflater.inflate(R.layout.viewholder_shimmer_layout, parent, false));
        io.supercharge.shimmerlayout.ShimmerLayout layout = (ShimmerLayout) itemView;

        View innerView = inflater.inflate(innerViewResId, layout, false);
        layout.addView(innerView);
    }

    /**
     * Binds the view
     */
    public void bind() {

        ShimmerLayout layout = (ShimmerLayout) itemView;
        layout.startShimmerAnimation();
    }
}
```

**Comment by veghtomi** [comment]
@sharish maybe could you update the library to use the mentioned shimmer effect? Or if you do not have time, I would be happy to do it!

**Comment by sharish** [comment]
@veghtomi : Thanks for giving your inputs regarding the issue. I havent looked into this issue yet and  may need little time to see where memory issue raises. Since you are the author of https://github.com/team-supercharge/ShimmerLayout, Can you tell me where the facebook shimmer lags?

Thanks,
Harish

**Comment by veghtomi** [comment]
@sharish the main problem is that the facebook library uses too much ram if the number or the size of the views which are animated is big. To achieve the effect it creates 3 bitmap with ARGB_8888 config. I also wrote an article about it where you can find the details: https://medium.com/supercharges-mobile-product-guide/shimmerlayout-26978ab53c28

I think the easiest solution would be to use this as a dependency instead of copying the concrete class.

**Comment by sharish** [comment]
Thanks for the article and for the insights about the memory pattern. I can add your's/your team's library as dependency and will remove facebook's shimmer. I might take few weeks as I have little availability for this repo. Meanwhile, If you are okay forking and sending a PR, I'm happy to integrate.

Thanks,
Harish

**Comment by veghtomi** [comment]
@sharish I created a pull request which solves this issue :)

**Comment by sharish** [comment]
The issue will be closed with the release v1.1. Thanks @veghtomi for your PR.

**Comment by sharish** [comment]
v1.1 is released. Closing the issue. If you still facing the performance issues, open a new issue.

Thanks,
Harish
